Source/Utility/Diagnostics/DiagProbe.cpp

```diff
@@ -0,0 +1,324 @@
+#include "DiagProbe.H"
+#include <AMReX_VisMF.H>
+#include <AMReX_FPC.H>
+#include <AMReX_PlotFileUtil.H>
+#include <regex>
+#include <cstdio>
+
+AMREX_GPU_DEVICE
+AMREX_FORCE_INLINE
+amrex::Real
+LinearInterpolate(
+  amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> xp,
+  amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> x_low,
+  amrex::Array3D<amrex::Real, 0, 1, 0, 1, 0, 1> cell_data,
+  amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> dx)
+{
+  amrex::Real alpha, beta, gama;
+  amrex::Real value = 0.0;
+
+  alpha = 0.0;
+  beta = 0.0;
+  gama = 0.0;
+
+  const int XDIR = 0;
+  const int YDIR = 1;
+  const int ZDIR = 2;
+
+  alpha = (xp[XDIR] - x_low[XDIR]) / dx[XDIR];
+  if (AMREX_SPACEDIM >= 2) {
+    beta = (xp[YDIR] - x_low[YDIR]) / dx[YDIR];
+  }
+  if (AMREX_SPACEDIM == 3) {
+    gama = (xp[ZDIR] - x_low[ZDIR]) / dx[ZDIR];
+  }
+
+  value += (1.0 - alpha) * (1 - beta) * (1 - gama) * cell_data(0, 0, 0);
```

Comment:
add some dimensionality checks here so the unused pieces for lower dimensions don't get called

Reply:
For low-dimensional cases, the third index values in cell_data will be zeros, contributing zero values to the interpolation.